finance-service/src/main/kotlin/com/ajaxproject/financeservice/service/FinanceService.kt

```diff
@@ -33,16 +34,30 @@ class FinanceService(
     }
 
     fun getCurrentBalance(userId: Long): Mono<Double> {
-        return Flux.zip(
-            getAllFinancesByUserId(userId, INCOME).collectList(),
-            getAllFinancesByUserId(userId, EXPENSE).collectList()
+        return Mono.zip(
+            getAllIncomesByUserId(userId),
+            getAllExpensesByUserId(userId)
         )
-            .map { (incomes, expenses) ->
-                incomes.sumOf { it.amount } - expenses.sumOf { it.amount }
-            }
-            .switchIfEmpty(0.0.toMono())
-            .next()
+            .map { (incomes, expenses) -> incomes - expenses }
     }
+
+    private fun getAllIncomesByUserId(userId: Long): Mono<Double> {
+        return financeRepositoryImpl.findByUserIdAndFinanceType(userId, INCOME)
+            .collectList()
+            .map { incomes -> incomes.sumOf { it.amount } }
+            .switchIfEmpty { 0.0.toMono() }
+    }
+
+    private fun getAllExpensesByUserId(userId: Long): Mono<Double> {
+        return financeRepositoryImpl.findByUserIdAndFinanceType(userId, EXPENSE)
+            .collectList()
+            .map { incomes -> incomes.sumOf { it.amount } }
+            .switchIfEmpty { 0.0.toMono() }
+    }
+}
+
+fun String?.toUnknownError(): String {
```

Comment:
does naming correspond to actual behavior? 
and is it related to finance that you put it to finance service?

Reply:
I think this is a bad solution, so I will try the option below